contrib/epee/include/net/net_helper.h

```diff
@@ -240,7 +270,10 @@ namespace net_utils
 				// object is used as a callback and will update the ec variable when the
 				// operation completes. The blocking_udp_client.cpp example shows how you
 				// can use boost::bind rather than boost::lambda.
-				boost::asio::async_write(m_socket, boost::asio::buffer(buff), boost::lambda::var(ec) = boost::lambda::_1);
+				if(!m_ssl)
+					boost::asio::async_write(m_ssl_socket.next_layer(), boost::asio::buffer(buff), boost::lambda::var(ec) = boost::lambda::_1);
```

Comment:
The docs weren't very clear on this. At first I used lowest_layer for all operations except reads and writes after reading this https://stackoverflow.com/a/32584870. Now i've changed to only using next_layer because I don't really see the benefit in using both.